Rubberduck.Parsing/VBA/RubberduckParserState.cs

```diff
@@ -71,17 +71,8 @@ public sealed class RubberduckParserState : IDisposable, IDeclarationFinderProvi
 
         public RubberduckParserState(IVBE vbe, IDeclarationFinderFactory declarationFinderFactory)
         {
-            if(vbe == null)
-            {
-                throw new ArgumentNullException(nameof(vbe));
-            }
-            if (declarationFinderFactory == null)
-            {
-                throw new ArgumentNullException(nameof(declarationFinderFactory));
-            }
-
-            _vbe = vbe;
-            _declarationFinderFactory = declarationFinderFactory;
+            _vbe = vbe ?? throw new ArgumentNullException(nameof(vbe));
+            _declarationFinderFactory = declarationFinderFactory ?? throw new ArgumentNullException(nameof(declarationFinderFactory));
```

Comment:
Should we be crashing runtime like that?

Reply:
this is just c#7-ification of existing behaviour...

Reply:
I really do not like the change here. This way, it is rather hard to see the guard clauses; it no longer screams: "I want non-null arguments!"

Reply:
@MDoerner to me that seems to just be a matter of getting used to it. I can see benfits of both approaches. The one is significantly more explicit, the other conserves screen real-estate...